syntax = "proto2";

package Codec_LedgerEntry;

message LedgerEntry
{
	required bytes ID = 1;
	required sint64 balance = 2;
	
	optional uint32 txHeight = 3;
	optional bytes txHash = 4;
	optional uint32 index = 5;
	optional fixed32 txtime = 6;
	
	optional bool isCoinbase = 7;
	optional bool isChangeBack = 8;
	optional bool isSTS = 9;
	optional bool optInRBF = 10;
	optional bool isChainedZC = 11;
	optional bool isWitness = 12;

	repeated bytes scrAddr = 13;
}

message ManyLedgerEntry
{
	repeated LedgerEntry values = 1;
}